controllers/main.js

```diff
@@ -123,28 +125,43 @@ module.exports.renderPlay = async (req, res) => {
 module.exports.renderResults = async (req, res, next) => {
   const { id } = req.params;
   let playSessionData = await PlaySession.findOne({ sessionId: id });
-  const highscoreText = `${
-    remapDifficultyTexts[playSessionData.difficulty] + playSessionData.length
-  }`;
-  const difficultyFilter = `bestScores.${playSessionData.difficulty}`;
-  const highscoresData = JSON.stringify(
-    await User.find()
-      .sort({ [difficultyFilter]: -1 })
-      .limit(10)
-  );
 
   playSessionData = JSON.stringify(playSessionData);
   res.render("main/results", {
     extraStyles: resultsStyle + mediaButtonsStyle,
     playSessionData,
-    highscoresData,
-    highscoreText,
-    defaultProfileImg: playSettingsImage,
     sessionId: id,
   });
   next();
 };
 
+module.exports.fetchTopHighscores = async (req, res) => {
+  const { difficulty, length } = req.query;
+  const dbSearchFilter = `bestScores.${difficulty}.${length}`;
+  async function getTop10Users() {
```

Comment:
Two things I'd like to comment on. First I would prefer if this function would be moved outside `fetchTopHighscores` to create reusable modules of functions (even if it's not needed at the moment) to keep it strict and clean. You can declare `getTop10Users` function somewhere outside and have it accept a `dbSearchFilter` parameter.

And second point, try to stay consistent with declarations of functions, choose one method and stick to it. These days it's more common to store a function inside a variable like so

```
  const getTop10Users = async () => {
   ...
  };
```

Reply:
You commented on the getTop10Users function in the previous PR as well. I think the code is highlighted in github because I did something wrong with rebasing, so that is show changes from previous PRs as well. It should be addressed in the other fix that I pushed today, and should be solved after another rebase.

As for the inconsistency in code, I believe it is mostly present because of the examples I used as i researched whenever I ran into issues, never really considered sticking to a single standard when defining them. Your suggested way of defining functions is something relatively new to me, although it is commonly used in the react course I'm going over now, and am quickly getting used to it. Will try to be more consistent with this in the future.